packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/ios/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,523 @@
+        } else {
+            // Select an authentictor and request the next step.
+            signInController.selectAuthenticator(authenticator: selectedOption,
+                                                 completion: completion)
+        }
+    }
+}
+
+```
+
+The UI updates the sign-on controller with the selected authenticator by calling `selectAuthenticator(authenticator:completion)`, or `skipAuthenticatorSelection(completion:)` if the user chose to skip enrolling in additional authenticators:
+
+```swift
+public func skipAuthenticatorSelection(completion:@escaping (Result<SuccessResultType, LoginError>) -> Void) {
+    // Set the current view model completion handler which is
+    // called when recieving a response from the SDK.
+    self.completion = completion
+
+    guard let skipRemediation = currentResponse?.remediations.filter({ $0.type == .skip }).first else {
+        // Handle the error
+        return
+    }
+
+    // Request the next step.
+    skipRemediation.proceed()
+}
+
+
+public func selectAuthenticator(authenticator: Remediation.Form.Field?, completion:@escaping (Result<SuccessResultType, LoginError>) -> Void) {
+    // Set the selected authenticator.
+    self.completion = completion
+
+    guard let selectorFields = authenticatorOptions() else {
+              self.finish(with: .cannotProceed)
+              return
+          }
+
+    // Update the remediation and send it to the server,
+    // requesting the next step.
+    selectorFields.optionSelectorField.selectedOption = authenticator
+    currentRemediation!.proceed()
+}
+```
+
+### 5: Display the shared secret, QR Code, and request the code
+
+After selecting the authenticator, the server sends an enrollment request in a remediation of type `.enrollAuthenticator`. Enrolling uses a secret key shared between Okta and Google Authenticator that's used to generate and verify the generated time-based one-time passcodes (TOTP).
+
+The remediation includes the shared secret as both a QR Code and as text. It also includes a request to enter a Google Authenticator code after enrollment.
+
+The sign-on controller code for `handleEnrollAuthenticator(remdiation:)` works for many different authenticators because the `form` property of the remediation contains the information required to build the UI:
+
+```swift
+func handleEnrollAuthenticator(_ remdiation: Remediation) {
+    // Tell the caller to present a UI for enrolling in Google Authenticator.
+
+}
+```
+
+The collection of authenticators related to the current request are in the `authenticators` property of the remediation. Each one is an instance of the `Authenticator` class which includes an identifier in the `key` property and a name for display in the UI in the `displayName` property.
+
+Authenticators may include one or more capabilities that describe associated data and behaviors, such as the QR Code or the ability to send a code. This includes
```

Comment:
Delete "This includes" or complete the sentence.

Reply:
Nice catch... deleted. That was part of a previous version as I worked through how to phrase things.

Reply:
nice